### ISSUE TITLE
Return promises if callback not present

### ISSUE THREAD

**Issue by IBMCode** [pwmckenna/node-travis-ci#25]
Would be great if this library return promises if a callback is not present for apis.

**Comment by ljharb** [comment]
At this point, it'd be better to make a breaking change to only return a promise, rather than accepting a callback at all.

**Comment by IBMCode** [comment]
Yeah, just for backward compatibility I suppose. For the time being, I am just wrapping mine in a promise. Bluebird promisify didn't work for me, had to do it manually.

**Comment by ljharb** [comment]
https://npmjs.com/util.promisify is also helpful.
